### ISSUE TITLE
[12.0] Sentry addon trows an exception

### ISSUE THREAD

**Issue by minimedj** [OCA/server-tools#1499]
I installed sentry addon for my odoo 12.

Here is a link of installed addon: https://github.com/OCA/server-tools/tree/12.0/sentry

The problem is that it trows an exception during odoo start:

```
(odoo12) max@mysite-dev:~/odoo/odoo_config$ ./run_development.sh
/home/max/odoo/odoo_config
/home/max/odoo
Sentry responded with an error: maximum recursion depth exceeded while calling a Python object (url: https://logging.mysite.club/api/2/store/)
Traceback (most recent call last):
  File "/home/max/.virtualenvs/odoo12/lib/python3.6/site-packages/raven/transport/threaded.py", line 165, in send_sync
    super(ThreadedHTTPTransport, self).send(url, data, headers)
  File "/home/max/.virtualenvs/odoo12/lib/python3.6/site-packages/raven/transport/http.py", line 43, in send
    ca_certs=self.ca_certs,
  File "/home/max/.virtualenvs/odoo12/lib/python3.6/site-packages/raven/utils/http.py", line 66, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.6/urllib/request.py", line 526, in open
    response = self._open(req, data)
  File "/usr/lib/python3.6/urllib/request.py", line 544, in _open
    '_open', req)
  File "/usr/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/home/max/.virtualenvs/odoo12/lib/python3.6/site-packages/raven/utils/http.py", line 46, in https_open
    return self.do_open(ValidHTTPSConnection, req)
  File "/usr/lib/python3.6/urllib/request.py", line 1318, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/usr/lib/python3.6/http/client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/usr/lib/python3.6/http/client.py", line 964, in send
    self.connect()
  File "/home/max/.virtualenvs/odoo12/lib/python3.6/site-packages/raven/utils/http.py", line 38, in connect
    sock, ca_certs=ca_certs, cert_reqs=ssl.CERT_REQUIRED)
  File "/home/max/.virtualenvs/odoo12/lib/python3.6/site-packages/gevent/_ssl3.py", line 576, in wrap_socket
    ciphers=ciphers)
  File "/home/max/.virtualenvs/odoo12/lib/python3.6/site-packages/gevent/_ssl3.py", line 126, in __init__
    self._context.verify_mode = cert_reqs
  File "/usr/lib/python3.6/ssl.py", line 485, in verify_mode
    super(SSLContext, SSLContext).verify_mode.__set__(self, value)
  File "/usr/lib/python3.6/ssl.py", line 485, in verify_mode
    super(SSLContext, SSLContext).verify_mode.__set__(self, value)
  File "/usr/lib/python3.6/ssl.py", line 485, in verify_mode
    super(SSLContext, SSLContext).verify_mode.__set__(self, value)
  [Previous line repeated 290 more times]
RecursionError: maximum recursion depth exceeded while calling a Python object
['Starting Odoo Server']
```

What could the reason? How to fix it?

**Comment by moylop260** [comment]
@naglis @jeffery9 
Could you check it, please?

Is there any issue using python3.6 and workers>1?

**Comment by moylop260** [comment]
FYI https://github.com/getsentry/sentry/issues/3384#issuecomment-233584126

>Jay-Pi: I can confirm that it is resolved since 8.5.1

So, it is not related with the module itself

**Comment by moylop260** [comment]
BTW I could reproduce this issue using py3.6

# Summary

If you install the following package version:
 - gevent==1.2.0

The issue will be fixed

# Large history

I just found the following comment:
 - https://github.com/gevent/gevent/issues/1016#issuecomment-328530752

Where the version of gevent could be the guilty
The warning of the comment above was added in `gevent==20.9.0`

Installing this version package I have not reproduced anymore but I don't recommend change the official version required for odoo.

In fact, installing `python3 -m pip install gevent==1.1.2` the error is reproduced
But installing `python3 -m pip install gevent==1.5.0` the error is fixed too. (I don't know what is the lower version required here)

Let's try

Not reproduced:
 - 1.2a1
   
 - 1.2.0
 - 1.4.0
 - 1.5.0
 - 20.9.0

Reproduced:
  - 1.1.2

Running `git log v1.2a1..v1.1.2 --reverse`
And testing commit by commit in order to find the real fix using `git reset --hard SHA` and running `pip install .`

Reproduced:
 - gevent@07404cbe0837e2cb85e878340afbcfb922d05384
 - gevent@70d2149412b7afcb5542f433e1c81b0929e22309
 - gevent@00b173ccb8f077627eb81c98f18619ffbbe93b67
 - gevent@35e2ab56e0615645ec1f4998471d636b47daf283 (tag: v1.1.2)
    - It is weird maybe how it was compiled it?
    -   I sus the following change but not sure: https://github.com/gevent/gevent/commit/5646ae7f024e966666f57dc034b1c3a2f515b63f

So the lower version is `gevent==1.2.0`

In fact, Odoo is installing a newer version for py3.7 since that original one was thinking in py3.5
So, maybe a particular version is required for py3.6 too in the odoo's requirements.txt
